### ISSUE TITLE
 Clarification on Number of Epochs forCLIP-Lora Model Training

### ISSUE THREAD

**Issue by marcomistretta** [MaxZanella/CLIP-LoRA#10]
Hi @MaxZanella,

I hope this message finds you well.

First of all, I want to thank you for your impressive work and for making the code publicly available. It has been very helpful in my research.

I was trying to reproduce your results but had a question regarding the training process. Specifically, I'm unsure about the statement:

> "The number of iterations is set equal to 500 times N/K (the number of labeled samples per class)"

Does this mean CLIP-Lora is trained for 8000 epochs (16 shots for each class) on each dataset? It seems like the model tends to overfit with so many epochs, especially on certain datasets.

Thank you in advance for your assistance.

Best regards,  
Marco

**Comment by MaxZanella** [comment]
Hi @marcomistretta, 

Thanks for your feedback, we really value it!

It seems there might be a misconception between epochs and iterations. Many works use epochs (one sweep over the whole dataset), but in this work, we found it more convenient to refer to the number of iterations (one iteration being one gradient descent step). We found iterations to be a more consistent hyperparameter, especially for small datasets like EuroSAT, where one epoch boils down to only one gradient descent step (in the very low-shot setting).

You can see in lora.py line 73, the number of iterations is computed as 

`
total_iters = args.n_iters * args.shots
`

with n_iters = 500 by default.

 Thanks for pointing this out. I agree it should be more explicit, and I will add a clarification in the code.

**Comment by marcomistretta** [comment]
Hi @MaxZanella,

Thank you for your clarification!
